drivers/backup/portworx/portworx.go

```diff
@@ -204,6 +206,37 @@ func (p *portworx) DeleteBackup(req *api.BackupDeleteRequest) (*api.BackupDelete
 	return p.backupManager.Delete(context.Background(), req)
 }
 
+// WaitForBackupCompletion waits for backup to complete successfully
+// or till timeout is reached. API should poll every `timeBeforeRetry` duration
+func (p *portworx) WaitForBackupCompletion(req *api.BackupInspectRequest,
+	timeout time.Duration, timeBeforeRetry time.Duration) error {
+	f := func() (interface{}, bool, error) {
+		inspectBkpResp, err := p.backupManager.Inspect(context.Background(), req)
+		if err != nil {
+			// Error occured, just retry
+			return nil, true, err
+		}
+
+		// Check if backup status is complete
+		currentStatus := inspectBkpResp.GetBackup().GetStatus().GetStatus()
+		if currentStatus == api.BackupInfo_StatusInfo_Success {
+			// If backup is complete, dont retry again
+			return nil, false, nil
```

Comment:
Stop retrying on failure and other terminal status

Reply:
@disrani-px done